lxd/container_lxc.go

```diff
@@ -1895,19 +1911,36 @@ func (c *containerLXC) startCommon() (string, error) {
 			// Unix device
 			paths, err := c.createUnixDevice(fmt.Sprintf("unix.%s", k), m)
 			if err != nil {
-				return "", err
+				if m["required"] == "" || shared.IsTrue(m["required"]) {
+					return "", err
+				}
+
```

Comment:
Probably worth a one line comment here to explain what's going on.

    // Deal with device hotplug

Reply:
added